### PR TITLE
Linting + include "extra version info" in all commands not just docker compose

### PR DESCRIPTION
**Pull request by achidlow**
Some other minor refactoring which was to reduce complexity pointed out by new linting rules

### CODE REVIEW

File: src/algokit/core/atomic_write.py [hunk 1]
```diff
@@ -12,10 +13,8 @@ def atomic_write(file_contents: str, target_file_path: Path, mode: Literal["a",
     temp_file_path = target_file_path.with_suffix(f"{target_file_path.suffix}.algokit~")
     try:
         # preserve file metadata if it already exists
-        try:
+        with contextlib.suppress(FileNotFoundError):
```
Comment (robdmoore) [inline src/algokit/core/atomic_write.py:16]: Cool!

File: tests/doctor/test_doctor.test_unexpected_exception.approved.txt [hunk 2]
```diff
@@ -6,10 +6,10 @@ DEBUG: Running 'git --version' in '{current_working_directory}'
 DEBUG: git: git version 2.37.1 (Apple Git-137.1)
 DEBUG: Running 'python --version' in '{current_working_directory}'
 DEBUG: python: Python 3.10.0
-DEBUG: Unexpected error checking dependency: OH NO
+DEBUG: Failed to locate python: OH NO
```
Comment (robdmoore) [inline tests/doctor/test_doctor.test_unexpected_exception.approved.txt:9]: Nice

File: tests/doctor/test_doctor.test_unexpected_exception.approved.txt [hunk 3]
```diff
@@ -30,10 +30,8 @@ OS: Darwin-other-system-info
 docker: 20.10.21
 docker compose: 2.12.2
 git: 2.37.1
-python: Unexpected error checking dependency
-  RuntimeError: OH NO
-python3: Unexpected error checking dependency
-  RuntimeError: OH NO
+python: 3.10.0f (location: unknown)
```
Comment (robdmoore) [inline tests/doctor/test_doctor.test_unexpected_exception.approved.txt:33]: I don't understand this, if there is an exception how can it know the version?

Comment (achidlow) [reply tests/doctor/test_doctor.test_unexpected_exception.approved.txt:33]: Yeah I kept the test name but maybe I should change it. It's really just a failure in locating the binary with `which`.

File: src/algokit/core/atomic_write.py (hunk 1 above)
Comment (achidlow) [reply src/algokit/core/atomic_write.py:16]: Thanks ruff! It suggested this and I didn't even know it existed lol.

File: tests/doctor/test_doctor.test_unexpected_exception.approved.txt (hunk 3 above)
Comment (achidlow) [reply tests/doctor/test_doctor.test_unexpected_exception.approved.txt:33]: Test name + approval file name updated

**Review by robdmoore — APPROVED** [review]

**Pull request merged**